libsmtutil/CHCSmtLib2Interface.cpp

```diff
@@ -223,3 +225,289 @@ std::string CHCSmtLib2Interface::createHeaderAndDeclarations() {
 std::string CHCSmtLib2Interface::createQueryAssertion(std::string name) {
 	return "(assert\n(forall " + forall() + "\n" + "(=> " + name + " false)))";
 }
+
+class SMTLibTranslationContext
+	{
+		CHCSmtLib2Interface const& m_chcInterface;
+		std::map<std::string, SortPointer> knownVariables;
+
+		bool isNumber(std::string const& _expr)
```

Comment:
Good question. It turns out yes, because in SMT-LIB, unary minus is an operator, so negative number should be represented as `(- <number>)`.

I'll try invariant with a negative number, to check this case.